### ISSUE TITLE
Vic 3 false-positive warnings

### ISSUE THREAD

**Issue by HiddeLekanne** [amtep/ck3-tiger#130]
In Vic3 a != doesn't mean the other value has to be a bool or value, it can be any identifying scope.

Example
```
warning(scopes): `this` produces country but expected value or bool
   --> [MOD] common\scripted_effects\imperia_vienna_congress_effects.txt
755 |                 global_var:imperia_vienna_congress_session_target != this
    |                                                                      ^^^^ 
700 |     every_country = {
    |     ^^^^^^^^^^^^^ <-- scope was deduced from `every_country` here
```
    
   
The "special_selection" seems to just inherit the scope of it's outer scope. It now has country, country definition and power block as a use. They should all be able to be identified by the outer scope it is in.

Example:
```
warning(scopes): `power_bloc_leader` is for power bloc but scope seems to be country or country definition
   --> [MOD] common\coat_of_arms\template_lists\coa_templates.txt
417 |                 ROOT.power_bloc_leader = c:GBR
    |                      ^^^^^^^^^^^^^^^^^ 
414 |         special_selection = {
    |         ^^^^^^^^^^^^^^^^^ <-- scope was supplied by the game engine
```

**Comment by amtep** [comment]
The first one is very tricky. Tiger generally recognizes such comparisons, but not when it knows nothing about the left-hand side (a global_var could be anything). Tweaking that logic will require a good think.

The second one is easy enough to solve by adding PowerBloc as one of the options. Figuring out what the surrounding scope is is more difficult.

**Comment by amtep** [comment]
Fixed in 639c961d1cd6402ceb390cc50074f3ff006bf779
